pyhf/pdf.py

```diff
@@ -17,6 +17,7 @@ def __init__(self, spec, poiname='mu'):
         self.par_map = {}
         self.par_order = []
         self.next_index = 0
+        self.poi_name = None
```

Comment:
why is this data member needed? I don't see it being used anywhere in the PR

Reply:
This was added to make it easier to access the poi name instead of going through index -- mostly for other users. We do store poi_index, but it would be tricky for users to figure out how to get the poi name from that (and they don't necessarily need to know the internals of ModelConfig to do so).